framework/include/media/OutputDataSource.h

```diff
@@ -78,23 +89,88 @@ class OutputDataSource : public DataSource
 	 */
 	virtual ~OutputDataSource();
 
+public:
+	/**
+	 * @brief Puts the stream data
+	 * @details @b #include <media/OutputDataSource.h>
+	 * @since TizenRT v2.0 PRE
+	 */
+	virtual ssize_t write(unsigned char *buf, size_t size);
+
+	/**
+	 * @brief Register an observer to get data souce state and other infomations.
+	 * @details @b #include <media/OutputDataSource.h>
+	 * @since TizenRT v2.0 PRE
+	 */
+	void setObserver(std::shared_ptr<OutputDataSourceObserverInterface> observer) { mObserver = observer; }
+
+	/**
+	 * @brief Start data source worker.
+	 * @details @b #include <media/OutputDataSource.h>
+	 * @since TizenRT v2.0 PRE
+	 */
+	bool start();
```

Comment:
@zhouxinhe 
i have one question.
why is this api public?
Do Walker handling in app ~?